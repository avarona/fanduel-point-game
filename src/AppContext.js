import React from 'react';

export const AppContext = React.createContext();

export const AppContextProvider = ({ value, children }) => {
    return (
        <AppContext.Provider value={value}>
            {children}
        </AppContext.Provider>
    )
}
