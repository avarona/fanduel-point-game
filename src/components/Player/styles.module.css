.playerContainer {
    position: relative;
    display: flex;
    flex-direction: column;
    cursor: pointer;
    user-select: none;
    padding: 24px;
    box-shadow: 0px 1px 2px grey;
    border-radius: 16px;
    background: white;
    transition: box-shadow 0.3s ease-in-out;
}

.playerContainer:hover {
    box-shadow: 0 5px 15px rgba(0,0,0,0.3);
}

.image {
    width: 200px;
    height: 200px;
    transition: opacity 0.2s ease-in-out;
}

.name, .score {
    align-self: center;
    font-size: 18px;
}

.disabled {
    pointer-events: none;
    cursor: not-allowed;
}

.disabled.playerContainer:hover {
    box-shadow: 0 1px 2px grey;
}

.winnerSelected {
    opacity: .2;
}

.winnerIcon {
    opacity: 1;
    width: 100px;
    position: absolute;
    top: 30%;
    left: 30%;
}