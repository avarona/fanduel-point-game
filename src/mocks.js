export const players = [
    {id: 1, first_name: 'Kobe', last_name: 'Bryant', fppg: 15, images: { default: { url: 'slick pic' }}},
    {id: 2, first_name: 'Lebron', last_name: 'James', fppg: 10, images: { default: { url: 'the dunker' }}},
    {id: 3, first_name: 'Michael', last_name: 'Jordan', fppg: 5, images: { default: { url: 'the legend' } }}
];

export const matchPlayers = [players[0], players[1]];

export const playerDetails = players[0];