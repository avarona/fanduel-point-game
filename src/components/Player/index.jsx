import React from 'react';
import classnames from 'classnames';
import { concatFullName, roundNumber } from '../../utils';
import styles from './styles.module.css';
import { CHECK_ICON, X_ICON } from '../../constants';
import { AppContext } from '../../AppContext';

const WinOrFail = (winnerId, selectId) => (
    winnerId === selectId
    ? <img className={styles.winnerIcon} alt="check mark" src={CHECK_ICON} />
    : <img className={styles.winnerIcon} alt="x mark" src={X_ICON} />
)

const Player = ({ details }) => {
    const fullName = concatFullName(details.first_name, details.last_name);
    const score = roundNumber(details.fppg);
    const { url } = details.images.default;
    const { id } = details;

    return (
        <AppContext.Consumer>
            {({ selectPlayer, winnerId, selectId }) => (
                <div className={classnames(styles.playerContainer, { [styles.disabled]: winnerId })} onClick={() => selectPlayer(id)}>
                    <img 
                        className={classnames(styles.image, { [styles.winnerSelected]: winnerId })}
                        src={url}
                        alt={fullName}
                    />
                    <span className={styles.name}>{fullName}</span>
                    
                    {/* show score only when a winner is determined */}
                    {winnerId && <span className={styles.score}>{score}</span>}

                    {/* render an icon for the winner/loser */}
                    {selectId === id ? WinOrFail(winnerId, id) : null}
                </div>
            )}
        </AppContext.Consumer>
    )
};

export default Player;