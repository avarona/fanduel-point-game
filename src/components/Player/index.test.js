import React from 'react';
import { mount } from 'enzyme';
import Player from './';
import { AppContextProvider } from '../../AppContext';
import * as Mocks from '../../mocks';

describe('Player Component', () => {
    describe('with winnerId as `undefined`', () => {
        const context = { selectPlayer: jest.fn(), winnerId: undefined };
        const wrapper = mount(
            <AppContextProvider value={context}>
                <Player details={Mocks.playerDetails} />
            </AppContextProvider>
        );

        it('renders', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });

        it('selectPlayer is called when image is clicked', () => {
            wrapper.find('img').simulate('click');
            expect(context.selectPlayer).toHaveBeenCalled();
        });

    });

    describe('with winnerId as `1`', () => {
        const context = { selectPlayer: jest.fn(), winnerId: 1 };
        const wrapper = mount(
            <AppContextProvider value={context}>
                <Player details={Mocks.playerDetails} />
            </AppContextProvider>
        );

        it('renders', () => {
            expect(wrapper.html()).toMatchSnapshot();
        });
    });
});
