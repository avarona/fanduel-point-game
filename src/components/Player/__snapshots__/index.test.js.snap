// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player Component with winnerId as \`1\` renders 1`] = `"<div class=\\"playerContainer disabled\\"><img class=\\"image winnerSelected\\" src=\\"slick pic\\" alt=\\"Kobe Bryant\\"><span class=\\"name\\">Kobe Bryant</span><span class=\\"score\\">15.00</span></div>"`;

exports[`Player Component with winnerId as \`undefined\` renders 1`] = `"<div class=\\"playerContainer\\"><img class=\\"image\\" src=\\"slick pic\\" alt=\\"Kobe Bryant\\"><span class=\\"name\\">Kobe Bryant</span></div>"`;
